appservice/src/tree/DeploymentTreeItem.ts

```diff
@@ -144,6 +148,30 @@ export class DeploymentTreeItem extends AzureTreeItem<ISiteTreeRoot> {
         });
     }
 
+    public async viewCommitInGitHub(): Promise<void> {
+        // if it's not connected to a .git repo, this won't work
+        if (this.parent.contextValue === DeploymentsTreeItem.contextValueConnected) {
+            const sourceControl: SiteSourceControl = await this.root.client.getSourceControl();
+            const githubUrl: string = 'https://github.com';
+
+            if (sourceControl.repoUrl && sourceControl.repoUrl.match(githubUrl)) {
+                try {
+                    const githubCommitUrl: string = `${sourceControl.repoUrl}/commit/${this._deployResult.id}`;
+                    // tslint:disable-next-line:no-unsafe-any
+                    const githubPage: Response = await request({url: githubCommitUrl, resolveWithFullResponse: true}).promise();
+                    if (githubPage.statusCode === 200) {
```

Comment:
I'm hesitant to check the statusCode here. There's a risk that you'll block the user from viewing a commit in github that actually would've worked and I don't think the reward is worth it. If you don't check this they might get an error page on github saying "can't find this commit" but that seems fine to me